File: types/member.ts
export interface Member {
  id: string;
  name: string;
  phone: string;
  address?: string;
  city?: string;
  pincode?: string;
  proofId?: string;
  profilePic?: string | null;
  propertyId?: string;
  buildingId?: string;
  floorId?: string;
  roomId?: string;
  bedId?: string;
  bedAmount?: number;
  billingPeriod?: string;
  createdAt: string;
}

File: backend/app/models/member.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime

class MemberBase(BaseModel):
    name: str
    phone: str
    villageName: Optional[str] = None
    joinedDate: Optional[datetime] = None
    payDate: Optional[datetime] = None
    paymentCycle: Optional[int] = None
    nextDueDate: Optional[datetime] = None
    proofId: Optional[str] = None
    profilePic: Optional[str] = None
    propertyId: str
    buildingId: str
    floorId: str
    roomId: str
    bedId: str

class MemberCreate(MemberBase):
    pass

class MemberInDB(MemberBase):
    id: str
    createdAt: datetime

class MemberOut(MemberInDB):
    pass
